src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs

```diff
@@ -110,31 +126,19 @@ public void RequestHeader_QueryResponse_Fail()
         }
 
         [Fact]
-        public void NameValue_Add_Success()
-        {
-            WebHeaderCollection w = new WebHeaderCollection();
-            w.Add("Accept", "text/json");
-        }
-
-        [Fact]
-        public void NameValue_AddNullName_Fail()
+        public void Setter_ValidName_Success()
         {
             WebHeaderCollection w = new WebHeaderCollection();
-            Assert.Throws<ArgumentNullException>(() => w.Add(null, "test"));
+            w["Accept"] = "text/json";
         }
 
-        [Fact]
-        public void NameValue_AddEmptyName_Fail()
-        {
-            WebHeaderCollection w = new WebHeaderCollection();
-            Assert.Throws<ArgumentException>(() => w.Add("", "test"));
-        }
-
-        [Fact]
-        public void NameValue_AddEmptyValue_Success()
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void Setter_NullOrEmptyName_Throws(string name)
         {
             WebHeaderCollection w = new WebHeaderCollection();
-            w.Add("custom", "");
+            Assert.Throws<ArgumentNullException>(() => w[name] = "test");
```

Comment:
I updated the tests to validate the param name for the cases where it is specified in the implementation.

The implementation of `CheckBadChars` throws `ArgumentException` but does not currently specify the param name (although, it does include it in the message). I've left this as-is for now. I originally had split this method into two methods: one for validating name, the other for validating the value, and added the param names using `nameof(name)` and `nameof(value)`. But it made the diff harder to read and the full framework doesn't include the param name, so I held off on it for this PR. It'd be a nice follow-up improvement, though.